### DISCUSSION TITLE
Support for KSTAR KSG 5K DM

### DISCUSSION THREAD

**Discussion by Unkie-NL** [StephanJoubert/home_assistant_solarman#680] [Ideas]
I would like to use Solarman, but my inverter (KStar KSG 5K DM) isn't supported.
My serial number starts with IO 71653Dxxxxxx. The input field supports only integer.
Solarman is unable to connect to my inverter :-(...

Is this even possible or do I have to look for other alternatives?

**Comment by PV-Joe** [comment]
Did you try this one?

https://github.com/davidrapan/ha-solarman

**Reply by Unkie-NL** [reply]
Thanks, but it has the same problem.

My inverter serial number is alphanumeric. It consists of both letters and numbers. SolarMan expects an integer (only numbers). If I remove the letters, it doesn't work.

I am currently reading my data from the KStar KSG 5K DM with the SmartClient application (available on Android and iOS). On another forum they solved it by measuring the complete electrical installation, see [here](https://diysolarforum.com/threads/is-there-an-alternative-to-the-solar-smart-client-app-wifi-module.38331/).

Installation of KStar KSG 5K DM with SmartClient see [this ](https://www.youtube.com/watch?v=zHDH1neUuRA&ab_channel=VortexSolar) Youtube video.

**Reply by davidrapan** [reply]
As far as I know Solarman Stick Loggers with alphanumeric serials are "older" versions and are not (and won't be) supported with any integration which is using pysolarmanv5.

Are you sure that the serial is serial of the stick and not the inverter?

**Reply by Unkie-NL** [reply]
> w Solarman Stick

Thanks for your reply and attempt to help me.
Im sure its the serial of the stick and not the inverter. Is it possible to get a new Stick Logger for the KSTAR KSG 5K DM that is compatible with Home Assitant? 

Now that I write this question, I realize I have to ask this to KSTAR ;-)

**Reply by davidrapan** [reply]
I don't see any reason why that shouldn't be possible, for example: [LSW-3](https://www.solarmanpv.com/products/data-logger/stick-logger/). 😉

**Reply by Unkie-NL** [reply]
> I don't see any reason why that shouldn't be possible, for example: [LSW-3](https://www.solarmanpv.com/products/data-logger/stick-logger/). 😉

I thought maybe each vendor has their own protocols. Other words: Does de Wifi data logger understand the language the inverter is speaking.

**Reply by davidrapan** [reply]
Yes, some of them do but my integration (this one too) are implemented using solarman protocol. So if you want use our integration you need logger from solarman. And KSTAR is supported.
